Reorg tiering policy sections into manage tiering (#3524)
timescale/docs: use-timescale/data-tiering/enabling-data-tiering.md

You can query the data on the object storage tier, but you cannot modify it. Make sure that you are not tiering data that needs to be **actively modified**.

## Enable tiered storage

You enable tiered storage from the `Overview` tab in Console.

<Procedure>

1. **In [Timescale Console][console], select the service to modify**.

    You see the `Overview` section.

1. **Scroll down, then click `Enable tiered storage`**.

   ![Enable tiered storage](https://assets.timescale.com/docs/images/console-enable-tiered-storage.png)

   When tiered storage is enabled, you see the amount of data in the tiered object storage.

   <Highlight type="note">
   Data tiering is available in [Scale and Enterprise](/about/latest/pricing-and-account-management/) pricing plans only.
   </Highlight>

</Procedure>

## Automate tiering with policies

A tiering policy automatically moves any chunks that only contain data
older than the `move_after` threshold to the object storage tier. This works similarly to a
[data retention policy][data-retention], but chunks are moved rather than deleted. 

> A tiering policy schedules a job that runs periodically to asynchronously migrate eligible chunks to object storage. Chunks are considered tiered once they appear in the `timescaledb_osm.tiered_chunks` view. 

Reviewer: ```suggestion
A tiering policy schedules a job that runs periodically to asynchronously migrate eligible chunks to object storage.  After chunks are tiered, they appear in the `timescaledb_osm.tiered_chunks` view. 
```

---
title: Manage tiering
excerpt: How to enable and use object storage tiering
products: [cloud]
keywords: [tiered storage]
tags: [storage, data management]
cloud_ui:
    path:
        - [services, :serviceId, overview]
plans: [scale, enterprise]  
---

> # Manage tiering

Reviewer 2: Maybe the title should explain more clearly what we explain. Manage automatic and manual tiering?

You use tiered storage to save on storage costs. Specifically, you can migrate rarely used data from Timescale's standard high-performance storage to the object storage. After you enable tiered storage, you then either create automated tiering policies or manually tier and untier data.

> <Highlight type="note">

Reviewer 2: Can you move this to under `When tiered storage is activated` please

Author: moved. does not make sense to me to give the steps first and only explain why it does not work as an afterthought

> You use tiered storage to save on storage costs. Specifically, you can migrate rarely used data from Timescale's standard high-performance storage to the object storage. After you [enable tiered storage](#enable-tiered-storage), you then either [create automated tiering policies](#automate-tiering-with-policies) or [manually tier and untier data](#manually-tier-and-untier-chunks).

Reviewer 2: ```suggestion
You use tiered storage to save on storage costs. Specifically, you can migrate rarely used data from Timescale's standard high-performance storage to object storage. After you [enable tiered storage](#enable-tiered-storage), you then either [create automated tiering policies](#automate-tiering-with-policies) or [manually tier and untier data](#manually-tier-and-untier-chunks).
```

Data tiering is available in [Scale and Enterprise](/about/latest/pricing-and-account-management/) pricing plans only.

</Highlight>

Data on the object storage tier cannot be modified - so inserts, updates, and deletes will not work on tiered data. Make sure that you are not tiering data that is being <b>actively modified</b> to the object storage tier.

## Enable tiered storage

You enable tiered storage from the `Overview` tab in Console.

<Procedure>

> 1. In [Timescale Console][console], select the service to modify.

Reviewer 2: Bold for the steps please.

    You see the `Overview` section.

1. Scroll down, then click `Enable tiered storage`.

   ![Enable tiered storage](https://assets.timescale.com/docs/images/console-enable-tiered-storage.png)

   When tiered storage is activated, you see the amount of data in the tiered object storage.

</Procedure>

## Automate tiering with policies

A tiering policy automatically moves any data chunks that only contain data
older than the `move_after` threshold to the object storage tier. This works similarly to a

> [data retention policy][data-retention], but chunks are moved rather than deleted. You can add tiering policies to hypertables, including continuous aggregates.

Reviewer 2: Can you add links to hypertables and continuous aggregates please.

A tiering policy schedules a job that runs periodically to asynchronously migrate eligible chunks to object storage. Chunks are considered tiered once they appear in the `timescaledb_osm.tiered_chunks` view. Tiering does not influence your ability to query the chunks.

### Add a tiering policy

To add a tiering policy, connect to your $SERVICE_SHORT and call `add_tiering_policy`:

```sql
SELECT add_tiering_policy(hypertable REGCLASS, move_after INTERVAL, if_not_exists BOOL = false);
```

> For example, tier chunks that are more than three days old in the `example` hypertable in the following way: 

Reviewer 2: ```suggestion
For example, to tier chunks that are more than three days old in the `example` hypertable: 
```

I know, it is not super google friendly

Author: Well exactly

```sql
SELECT add_tiering_policy('example', INTERVAL '3 days');
```

### Remove a tiering policy

To remove an existing tiering policy, call `remove_tiering_policy`:

```sql
SELECT remove_tiering_policy(hypertable REGCLASS, if_exists BOOL = false);
```

> For example, remove the tiering policy from the `example` hypertable in the following way:

Reviewer 2: ```suggestion
For example, to remove the tiering policy from the `example` hypertable:
```

   than three days:

   ```sql
   SELECT show_chunks('example', older_than => INTERVAL '3 days');
   ```

    This returns a list of chunks. Take a note of the chunk names:

   ```sql
   |1|_timescaledb_internal_hyper_1_2_chunk|
   |2|_timescaledb_internal_hyper_1_3_chunk|
   ```

1. Call the `tier_chunk` function to manually tier each chunk:

   ```sql
   SELECT tier_chunk( '_timescaledb_internal_hyper_1_2_chunk');
   ```

1. Repeat for all chunks you want to tier.

1. To see which chunks are tiered into the object storage tier, use the `tiered_chunks` informational view:

    ```sql
    SELECT * FROM timescaledb_osm.tiered_chunks;
    ```

</Procedure>

Tiering a chunk schedules it for migration to the object storage tier, but the migration won't happen immediately. Chunks are tiered one at a time in order to minimize database resource consumption. You can continue to query a chunk during migration.

To see which chunks are scheduled for tiering either by policy or by a manual call, but have not yet been tiered, use this view:

```sql
SELECT * FROM timescaledb_osm.chunks_queued_for_tiering ;
```

### Untier chunks

> Tiered data is immutable. To update data in a tiered chunk, move it back to local storage, that is, Timescale's standard high-performance storage tier. You can do so by using the `untier_chunk` stored procedure.

Reviewer 2: storage. That is, 

The storage tier is in Timescale Cloud, Timescale is a company. 

To untier a chunk, call the `untier_chunk` stored procedure.

> Untiering chunks is a synchronous process. Chunks are renamed when the data is untiered.

Reviewer 2: Don't see this should be on a newline. 

Can you add a lead in sentence before the procedure please.

    public            | sample          | _hyper_1_1_chunk | 2023-02-16 00:00:00+00 | 2023-02-23 00:00:00+00
    (1 row)
    ```

1.  Run `untier_chunk`:

    ```sql
    CALL untier_chunk('_hyper_1_1_chunk');
    ```

1.  See the details of the chunk with the
    `timescaledb_information.chunks` function:

    ```sql
    SELECT * FROM timescaledb_information.chunks;
    ```

    The output looks something like this:

    ```sql
    -[ RECORD 1 ]----------+-------------------------
    hypertable_schema      | public
    hypertable_name        | sample
    chunk_schema           | _timescaledb_internal
    chunk_name             | _hyper_1_4_chunk
    primary_dimension      | ts
    primary_dimension_type | timestamp with time zone
    range_start            | 2023-02-16 00:00:00+00
    range_end              | 2020-03-23 00:00:00+00
    range_start_integer    |
    range_end_integer      |
    is_compressed          | f
    chunk_tablespace       |
    data_nodes             |
    ```

</Procedure>

## Disable tiering 

If you no longer want to use tiered storage for a particular hypertable, you

> can drop the associated metadata by calling the `disable_tiering` function.

Reviewer 2: hypertable, drop the ...

<Procedure>

1. Call `remove_tiering_policy` and drop any tiering policy associated with this hypertable.

1. Make sure that there is no tiered data associated with this hypertable:

    1. List the tiered chunks associated with this hypertable:
   
       ```sql
       select * from timescaledb_osm.tiered_chunks 
       ```

    1. If you have any tiered chunks, either untier this data, or drop these chunks from tiered storage.

1. Use `disable_tiering` to drop all tiering-related metadata for the hypertable:

   ```sql
   select disable_tiering('my_hypertable_name');
   ```

1. Verify that tiering has been disabled by listing the hypertables that have tiering enabled:

   ```sql
   select * from timescaledb_osm.tiered_hypertables;
   ```

</Procedure>

[data-retention]: /use-timescale/:currentVersion:/data-retention/

> [console]: https://console.cloud.timescale.com/

Reviewer 2: Can you update to https://console.cloud.timescale.com/dashboard/services so it takes the user directly to the services page please.

        - [services, :serviceId, overview]
---

# Manage tiering

> You use tiered storage to save on storage costs. Specifically, you can migrate rarely used data from Timescale's standard high-performance storage to the object storage. With tiered storage enabled, you then either create automated tiering policies or manually tier and untier data.

Reviewer 2: ```suggestion
You use tiered storage to save on storage costs. Specifically, you can migrate rarely used data from Timescale's standard high-performance storage to the object storage. After you enable tiered storage, you then either create automated tiering policies or manually tier and untier data.
```

> Data on the object storage tier cannot be modified - so inserts, updates, and deletes will not work on tiered data. Make sure that you are not tiering data that is being <b>actively modified</b> to the object storage tier.

Reviewer 2: Add a not saying this is for the Scale and Enterprise tiers only. Andddddd, please use the tiering widget in the page. Yaya

## Enable tiered storage

You enable tiered storage from the `Overview` tab in Console.

<Procedure>

> 1.  In Timescale Console, from the `Services` list, click the name of

Reviewer 2: ```suggestion
1.  In [Timescale Console][services-portal], select the service to modify. 
      You see the the Overview section.  
       
1. Scroll down, then click `Enable tiered storage`.
   ![Enable tiered storage](https://assets.timescale.com/docs/images/console-enable-tiered-storage.png)
    When tiered storage is activated, you see the amount of data in the tiered object storage.  
   

```

[services-portal]: https://console.cloud.timescale.com/dashboard/services

    the service you want to modify.

> 1.  In the `Overview` tab, locate the `Tiered Storage` card and click

Reviewer 2: remove the rest

</Procedure>

## Automate tiering with policies

> A tiering policy automatically moves data to the object storage tier. Any chunks that only contain data

Reviewer 2: A tiering policy automatically moves any  data chunks that only contain data
older than the `move_after` threshold to the object storage tier.

older than the `move_after` threshold are moved. This works similarly to a
[data retention policy][data-retention], but chunks are moved rather than deleted. You can add tiering policies to hypertables, including continuous aggregates.

> A tiering policy schedules a job that runs periodically to migrate eligible chunks. The migration is asynchronous. The chunks are considered tiered once they appear in the `timescaledb_osm.tiered_chunks` view. Tiering does not influence your ability to query the chunks.

Reviewer 2: ```suggestion
A tiering policy schedules a job that runs periodically to asynchronously migrate eligible chunks to object storage. Chunks are considered tiered once they appear in the `timescaledb_osm.tiered_chunks` view. Tiering does not influence your ability to query the chunks.
```

### Add a tiering policy

> To add a tiering policy, call `add_tiering_policy`:

Reviewer 2: ```suggestion
To add a tiering policy, connect to your ${SERVICE_SHORT} and call `add_tiering_policy`:

 ![Add a tiering policy](https://assets.timescale.com/docs/images/console-add-tiering-policy.png)
```

> ```sql

Reviewer 2: remove

Author: remove what?

Reviewer 2: The bits that are moved to the updates i suggested previously.

SELECT add_tiering_policy(hypertable REGCLASS, move_after INTERVAL, if_not_exists BOOL = false);
```

For example, tier chunks that are more than three days old in the `example` hypertable in the following way: 

```sql
SELECT add_tiering_policy('example', INTERVAL '3 days');
```

### Remove a tiering policy

To remove an existing tiering policy, call `remove_tiering_policy`:

```sql
SELECT remove_tiering_policy(hypertable REGCLASS, if_exists BOOL = false);
```

For example, remove the tiering policy from the `example` hypertable in the following way:

```sql
SELECT remove_tiering_policy('example');
```

> If you remove a tiering policy, new scheduled chunks will not be tiered. However, already tiered chunks won't be untiered. You can [untier chunks manually](#manually-tier-and-untier-chunks) to the local storage.

Reviewer 2: ```suggestion
If you remove a tiering policy, new scheduled chunks are not be tiered. However, chunks in tiered storage are not untiered. You [untier chunks manually](#manually-tier-and-untier-chunks) to local storage.
```

Author: This suggestion reverses the meaning

Reviewer 2: You are right, updated.

## Manually tier and untier chunks

If tiering policies do not meet your current needs, you can tier and untier chunks manually. 

### Tier chunks

> Tiering a chunk is an asynchronous process that schedules the chunk to be tiered. In this example, you use a hypertable called `example` and tier chunks older than three days. You then proceed to list tiered chunks.

Reviewer 2: in the following example, you tier chunks older than three days in the `example` [hypertable](link to hypertables). You then list the ...

<Procedure>

> 1. At the psql prompt, select all chunks in the table `example` that are older

Reviewer 2: Connect to your service using console or psql

or something along those lines. 

example hypertable

   than three days:

   ```sql
   SELECT show_chunks('example', older_than => INTERVAL '3 days');
   ```

    This returns a list of chunks. Take a note of the chunk names:

   ```sql
   |1|_timescaledb_internal_hyper_1_2_chunk|
   |2|_timescaledb_internal_hyper_1_3_chunk|
   ```

1. Call the `tier_chunk` function to manually tier each chunk:

   ```sql
   SELECT tier_chunk( '_timescaledb_internal_hyper_1_2_chunk');
   ```

1. Repeat for all chunks you want to tier.

1. To see which chunks are tiered into the object storage tier, use the `tiered_chunks` informational view:

    ```sql
    SELECT * FROM timescaledb_osm.tiered_chunks;
    ```

</Procedure>

> Tiering a chunk schedules it for migration to the object storage tier, but the migration won't happen immediately. Chunks are tiered one at a time in order to minimize database resource consumption. You can continue to query a chunk during migration.

Reviewer 2: Can you add this to the procedure please.

To see which chunks are scheduled for tiering either by policy or by a manual call, but have not yet been tiered, use this view:

```sql
SELECT * FROM timescaledb_osm.chunks_queued_for_tiering ;
```

### Untier chunks

> Tiered data is immutable. To update data in a tiered chunk, move it back to local storage, that is, Timescale's standard high-performance storage tier. You can do so by using the `untier_chunk` stored procedure.

Reviewer 2: To update data in a tiered chunk, move it back to the standard high-performance storage tier in ${CLOUD_LONG}.

Untiering chunks is a synchronous process. Chunks are renamed when the data is untiered.

<Procedure>

> 1.  At the `psql` prompt, check which chunks are currently tiered:

Reviewer 2: sql editor bla bla

    (1 row)
    ```

1.  Run `untier_chunk`:

    ```sql
    CALL untier_chunk('_hyper_1_1_chunk');
    ```

1.  See the details of the chunk with the
    `timescaledb_information.chunks` function:

    ```sql
    SELECT * FROM timescaledb_information.chunks;
    ```

    The output looks something like this:

    ```sql
    -[ RECORD 1 ]----------+-------------------------
    hypertable_schema      | public
    hypertable_name        | sample
    chunk_schema           | _timescaledb_internal
    chunk_name             | _hyper_1_4_chunk
    primary_dimension      | ts
    primary_dimension_type | timestamp with time zone
    range_start            | 2023-02-16 00:00:00+00
    range_end              | 2020-03-23 00:00:00+00
    range_start_integer    |
    range_end_integer      |
    is_compressed          | f
    chunk_tablespace       |
    data_nodes             |
    ```

</Procedure>

## Disable tiering 

If you no longer want to use tiered storage for a particular hypertable, you
can drop the associated metadata by calling the `disable_tiering` function.

<Procedure>

> 1. Call `remove_tiering_policy` and drop any tiering policy associated with this hypertable.

Reviewer 2: ```suggestion
SQL editor bla bla 

1. To drop all tiering policies associated with a table, call `remove_tiering_policy`.
```
